sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/FieldSingleDeclarationCheck.java

```diff
@@ -0,0 +1,120 @@
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import com.github.sevntu.checkstyle.Utils;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>This checks ensures that classes have at most 1 field of the given className.</p>
+ *
+ * <p>This can be useful for example to ensure that only one Logger is used:</p>
+ *
+ * <pre>
+ * &lt;module name="TreeWalker"&gt;
+ *   &lt;module name="com.github.sevntu.checkstyle.checks.coding.FieldSingleDeclarationCheck"&gt;
+ *       &lt;property name="className" value="org.slf4j.Logger" /&gt;
+ *   &lt;/module&gt;
+ * &lt;/module&gt;
+ *
+ * class Example {
+ *     private static final org.slf4j.Logger logger = LoggerFactory.getLogger(Example.class); // OK
+ *     private static       org.slf4j.Logger logger2; // NOK!
+ * }
+ * </pre>
+ *
+ * @author Milos Fabian, Pantheon Technologies - original author (in OpenDaylight.org)
+ * @author Michael Vorburger.ch - refactored and made more generalized for contribution to Sevntu
+ */
+public class FieldSingleDeclarationCheck extends AbstractCheck {
+
+    /**
+     * Violation message key.
+     */
+    public static final String MSG_KEY = "field.count";
+
+    /**
+     * Class name check for.
+     */
+    private String className;
+
+    /**
+     * Field to hold previously seen Logger.
+     */
+    private String prevSeenClassName = "";
+
+    /**
+     * Set Class of which there should only be max. 1 field per class.
+     * @param className the fully qualified name (FQN) of the class
+     */
+    public void setClassName(String className) {
+        this.className = className;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {TokenTypes.VARIABLE_DEF };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void beginTree(DetailAST rootAST) {
+        prevSeenClassName = null;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (isVariable(ast) && Utils.matchesFullyQualifiedName(ast, className)) {
```

Comment:
@rnveach you're right, I think (from what I understand of CS). But this almost more of a limitation of CS.. I had a quick look at EitherLogOrThrowCheck, and see how that Check has code to deal with this (hasLoggerClassInImports), but... that almost looks a like a one off hack to me. If you would like Checks to be able to be written for both full and local type names, then shouldn't there be better shared infrastructure available for writing such Checks? I would argue that mixed types of Logger is extremely atypical, and so while I agree this isn't completely perfect strictly speaking, it seems like a very minor problem in real world, and we should ignore this for now, and move forward. @romani your thoughts?